src/haddock/modules/scoring/emscoring/__init__.py

```diff
@@ -87,10 +91,8 @@ def _run(self):
         with open(output_fname, "w") as fh:
             fh.write(f"structure\toriginal_name\tscore{linesep}")
             for pdb in self.output_models:
-                # temporary solution to get the original name
-                #  here one .pdb = one .psf and the .psf is not changed by
-                #  the module so use its name as the original one
-                original_name = pdb.topology.file_name.replace('.psf', '.pdb')
+                # retrieve the original name
+                original_name = io_table[pdb.file_name]
                 fh.write(
                     f"{pdb.file_name}\t{original_name}\t{pdb.score}{linesep}"
```

Comment:
using `PDBFile.ori_name` instead:

```suggestion
                fh.write(
                    f"{pdb.file_name}\t{pdb.ori_name}\t{pdb.score}{linesep}"
```